test/test_script/test_import_struct_script.cc

```diff
@@ -0,0 +1,262 @@
+  }
+}
+
+TEST_F(ImportStructuredScriptTest, CreatesDukContextWithImportScript) {
+  EXPECT_NE(ctx, nullptr);
+}
+
+TEST_F(ImportStructuredScriptTest, AddsAudioItemWithABCBoxFormat) {
+  string title = "Audio Title";
+  string mimetype = "audio/mpeg";
+  string artist = "Artist";
+  string album = "Album";
+  string date = "2018-01-01";
+  string genre = "Genre";
+  string desc = "Description";
+  string id = "2";
+  string location = "/home/gerbera/audio.mp3";
+  int online_service = 0;
+  int theora = 0;
+  map<string, string> aux;
+
+  map<string, string> meta = {
+    make_pair("dc:title", title),
+    make_pair("upnp:artist", artist),
+    make_pair("upnp:album", album),
+    make_pair("dc:date", date),
+    make_pair("upnp:genre", genre),
+    make_pair("dc:description", desc)
+  };
+
+  // Represents the values passed to `addCdsObject`, extracted from keys defined there.
+  map<string, string> asAudioAllAudio = {
+    make_pair("title", title),
+    make_pair("meta['dc:title']", title),
+    make_pair("meta['upnp:artist']", artist),
+    make_pair("meta['upnp:album']", album),
+    make_pair("meta['dc:date']", date),
+    make_pair("meta['upnp:genre']", genre),
+    make_pair("meta['dc:description']", desc)
+  };
+
+  map<string, string> asAudioAllFullName = {
+    make_pair("title", "Artist - Album - Audio Title"),
+    make_pair("meta['dc:title']", title),
+    make_pair("meta['upnp:artist']", artist),
+    make_pair("meta['upnp:album']", album),
+    make_pair("meta['dc:date']", date),
+    make_pair("meta['upnp:genre']", genre),
+    make_pair("meta['dc:description']", desc)
+  };
+
+  map<string, string> asAudioAllArtistTitle = {
+    make_pair("title", "Audio Title (Album, 2018)"),
+    make_pair("meta['dc:title']", title),
+    make_pair("meta['upnp:artist']", artist),
+    make_pair("meta['upnp:album']", album),
+    make_pair("meta['dc:date']", date),
+    make_pair("meta['upnp:genre']", genre),
+    make_pair("meta['dc:description']", desc)
+  };
+  map<string, string> asAudioAllAudioTitleArtist = {
+    make_pair("title", "Audio Title - Artist"),
+    make_pair("meta['dc:title']", title),
+    make_pair("meta['upnp:artist']", artist),
+    make_pair("meta['upnp:album']", album),
+    make_pair("meta['dc:date']", date),
+    make_pair("meta['upnp:genre']", genre),
+    make_pair("meta['dc:description']", desc)
+  };
+  map<string, string> asAudioTrackArtistTitle = {
+    make_pair("title", "Audio Title - Artist (Album, 2018)"),
+    make_pair("meta['dc:title']", title),
+    make_pair("meta['upnp:artist']", artist),
+    make_pair("meta['upnp:album']", album),
+    make_pair("meta['dc:date']", date),
+    make_pair("meta['upnp:genre']", genre),
+    make_pair("meta['dc:description']", desc)
+  };
+
+  // Expecting the common script calls
+  // and will proxy through the mock objects
+  // for verification.
+  EXPECT_CALL(*commonScriptMock, getPlaylistType(Eq("audio/mpeg"))).WillOnce(Return(1));
```

Comment:
Some of the import operations create many, many CdsObject chains....lengthy testing for sure.